src/features/smartSearch/components/filters/EmailHistory/DisplayEmailHistory.tsx

```diff
@@ -0,0 +1,86 @@
+import convertToMessageKey from './convertToMessageKey';
+import DisplayTimeFrame from '../DisplayTimeFrame';
+import { EMAIL_SELECT_SCOPE } from '../EmailClick';
+import { getTimeFrameWithConfig } from '../../utils';
+import messageIds from 'features/smartSearch/l10n/messageIds';
+import { Msg } from 'core/i18n';
+import UnderlinedMsg from '../../UnderlinedMsg';
+import UnderlinedText from '../../UnderlinedText';
+import useCampaigns from 'features/campaigns/hooks/useCampaigns';
+import useEmails from 'features/emails/hooks/useEmails';
+import { useNumericRouteParams } from 'core/hooks';
+import {
+  EmailHistoryFilterConfig,
+  OPERATION,
+  SmartSearchFilterWithId,
+} from 'features/smartSearch/components/types';
+
+const localMessageIds = messageIds.filters.emailHistory;
+
+interface DisplayEmailHistoryProps {
+  filter: SmartSearchFilterWithId<EmailHistoryFilterConfig>;
+}
+
+const DisplayEmailHistory = ({
+  filter,
+}: DisplayEmailHistoryProps): JSX.Element => {
+  const { orgId } = useNumericRouteParams();
+  const emails = useEmails(orgId).data || [];
+  const projects = useCampaigns(orgId).data || [];
+
+  const { config } = filter;
+  const { operator, campaign: projectId, email: emailId } = config;
+
+  const op = filter.op || OPERATION.ADD;
+  const emailTitle = emails?.find((item) => item.id === emailId)?.title;
+  const projectTitle = projects?.find((item) => item.id === projectId)?.title;
+
+  const timeFrame = getTimeFrameWithConfig({
+    after: config.after,
+    before: config.before,
+  });
+
+  return (
+    <Msg
+      id={localMessageIds.inputString}
+      values={{
+        addRemoveSelect: <UnderlinedMsg id={messageIds.operators[op]} />,
+        emailScopeSelect: (
+          <UnderlinedMsg
+            id={
+              messageIds.filters.emailScopeSelect[
+                filter.config.campaign
+                  ? EMAIL_SELECT_SCOPE.FROM_PROJECT
+                  : filter.config.email
+                  ? EMAIL_SELECT_SCOPE.SPECIFIC_EMAIL
+                  : EMAIL_SELECT_SCOPE.ANY
+              ]
+            }
+          />
+        ),
+        emailSelect: emailId ? (
+          <UnderlinedText text={`"${emailTitle}"`} />
+        ) : null,
+        operatorSelect: (
+          <UnderlinedMsg
+            id={
+              localMessageIds.operatorSelect[
+                convertToMessageKey(operator) as
+                  | 'notSent'
+                  | 'opened'
+                  | 'sent'
+                  | 'notSent'
```

Comment:
I suggest replacing this with a look-up table in this module. I would put it outside of the component function, but that doesn't really matter. It could look something like this:

```tsx
const MESSAGE_KEY_BY_OP = {
  not_opened: 'notOpened',
  not_sent: 'notSent',
  opened: 'opened',
  sent: 'sent',
} as const;
```

Note the `as const` at the bottom, which should result in this being typed with explicit values, not just as any record.

Reply:
Thank you! I didn't know anything about `as const`! I really like it 😸!